azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -120,6 +124,19 @@ def __init__(self, auth_provider):
                 "dest": "connecting",
                 "after": "_call_provider_connect",
             },
+            {
+                "trigger": "_trig_disable_receive",
+                "source": "connected",
+                "dest": None,
+                "after": "_call_unsubscribe",
+            },
+            {"trigger": "_trig_disable_receive", "source": "connecting", "dest": None},
+            {
+                "trigger": "_trig_disable_receive",
+                "source": "disconnected",
+                "dest": "connecting",
+                "after": "_call_provider_connect",
+            },
```

Comment:
I don't know about that. triggering a connection just so that we can unsubscribe seems a little painful. I guess we could discuss the nuance of what the user is really trying to achieve (stop receiving messages vs actually cleaning up a subscription to a topic).

Reply:
I also felt this when I was trying to define the states this time. May be we should start thinking about the state machine in a different way. Bert was thinking through something. May be all subscriptions can happen during connection nd vice versa. As of now I will let this be.

---
In reply to: [258252555](https://github.com/Azure/azure-iot-sdk-python-preview/pull/50#discussion_r258252555) [](ancestors = 258252555)